lottie/src/main/java/com/airbnb/lottie/LottieDrawable.java

```diff
@@ -1244,7 +1245,8 @@ boolean isAnimatingOrWillAnimateOnVisible() {
   }
 
   private boolean animationsEnabled() {
-    return systemAnimationsEnabled || ignoreSystemAnimationsDisabled;
+    return (systemAnimationsEnabled || ignoreSystemAnimationsDisabled) &&
+        L.getReducedMotionOption().getCurrentReducedMotionMode(getContext()) == ReducedMotionMode.STANDARD_MOTION;
```

Comment:
If I'm reading this correctly, if you set `ignoreSystemAnimationsDisabled`  but the system scale is 0 then:
* You would expect animationsEnabled to return true
* systemAnimationsEnable: false
* ignoreSystemAnimationsDisabled: true
* LottieConfig: REDUCED

The predicate becomes (false || true) && (REDUCED == STANDARD) which would be false.

I think this may also read easier if you treat `ignoreSystemAnimationsDisabled == true` as higher priority with:
```
if (ignoreSystemAnimationsDisabled) {
    return true;
}
return L.getReducedMotionOption().getCurrentReducedMotionMode(getContext()) == ReducedMotionMode.STANDARD_MOTION;
```

Just off the top of my head, I think this is correct and easier to read. Is that right?

Reply:
you are right. Ideally `getCurrentReducedMotionMode` should not even be evaluated when `ignoreSystemAnimationsDisabled` is set to true. Let me fix it.

Reply:
I changed it to this 

```
private boolean animationsEnabled() {
    if (ignoreSystemAnimationsDisabled) {
      return true;
    }
    return systemAnimationsEnabled &&
        L.getReducedMotionOption().getCurrentReducedMotionMode(getContext()) == ReducedMotionMode.STANDARD_MOTION;
  }
  ```
  We still need to consider old `systemAnimationsEnabled` flag that can be set on drawable directly for backward compatibility